File: src/main/java/core/data/medimage/DataAttribute.java
package core.data.medimage;

/**
 * Stores a single tag that is part of metadata of medical images.
 */
public class DataAttribute<T> {

    public final String name;

    //TODO: (avlomakin) string for now, mb change to UUID
    public final String tag;

    public final T value;

    public String description;

    public DataAttribute(String name, String tag, T value) {
        this.name = name;
        this.tag = tag;
        this.value = value;
    }

    @Override
    public String toString() {
        return tag + " " + name + ": " + value.toString()  + ") ";
    }
}

File: src/main/java/core/data/attribute/DataAttribute.kt
package core.data.attribute

/**
 * Stores a single tag that is part of metadata of medical images.
 */
class DataAttribute<T>(val name: String, val tag: String, val value: T) : Cloneable {

    var description: String? = null

    public override fun clone(): DataAttribute<T> {
        return DataAttribute(name, tag, value)
    }

    override fun toString(): String {
        return tag + " " + name + ": " + value.toString() + ") "
    }
}

